public/js/services/pandaFetch.js

```diff
@@ -0,0 +1,39 @@
+import { reEstablishSession } from 'panda-session';
+import { getStore } from '../util/storeAccessor';
+
+function checkStatus(res) {
```

Comment:
This exists because unless there's a network error, fetch will resolve even with an error code.